decentraland/unity-renderer
# feat: Social rpc client

## What does this PR change?

This PR introduces the architecture  for a new RPC client for the [social service](https://github.com/decentraland/social-service), 
And it also introduces a feature flag (`use-social-client`) to prevent any unfinished from getting into production.

The solution also implied a few architectural changes in the project's codebase:
- FriendsController is now an IService and is registered in the ServiceLocator
  - All references were updated accordingly
- FriendsController is created receiving a `ISocialApiBridge` and receives a `ProxySocialApiBridge` that has the sole responsibility of knowing whether to call or not the RPC Client depending on the availability of the feature flag or not
- FriendsController now has an `InitializeAsync` and assigns the corresponding Actions for the friendships initialization the old bridge or the social one, depending on the feature flag.
- Only updated the friendsController that called the socialClient, like `RequestFriendshipAsync` and `RejectFriendshipAsync` as an example of how the client will be used in the future.

## How to test the changes?


1. Launch the explorer
2. Open the friends tab, the friends request
3. The current available friends and friend requests should still be there
4. The whole flow of a friend request should still work seamlessly (Send, is received in real-time, also canceling, and rejecting should work as expected)
5. When accepting or deleting a friend, the friend should be removed from the list of friends

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
